### ISSUE TITLE
Go client library cannot be used with dep

### ISSUE THREAD

**Issue by iamneal** [pachyderm/pachyderm#2983]
**What happened?**: 

looking at the [godoc](https://godoc.org/github.com/pachyderm/pachyderm/src/client)

I followed the example and created a `main.go` file in test package:
```s
mkdir $GOPATH/src/test_pach
touch $GOPATH/src/test_pach/main.go
cd $GOPATH/src/test_pach
```
I added these contents to the main.go file:
```go
package main

import (
	"github.com/pachyderm/pachyderm/src/client"
)

func main() {
	c, err := client.NewFromAddress("0.0.0.0:30650")
	if err != nil {
		panic(err)
	}
	println("we did it")
}
```
Here is [dep](https://github.com/golang/dep).  We use this tool to manage go dependencies.   I ran a: `dep init` command.  This generates a lock file for the project with all the transitive dependencies.

I then ran: `go build` 

**What you expected to happen?**:

I expected the minimal project to build.  Instead I got this error:
```
$ go build .
#<path_to_project>/vendor/github.com/pachyderm/pachyderm/src/client/pkg/config
vendor/github.com/pachyderm/pachyderm/src/client/pkg/config/config.go:36:13: assignment mismatch: 2 variables but 1 values
```
**How to reproduce it (as minimally and precisely as possible)?**:
paste the example above into a new go package/main.go, cd into that file and run
- `dep init` (takes awhile)
- `go build`

**Anything else we need to know?**:

**Environment?**:
```s
$ dep version
dep:
 version     : devel
 build date  :
 git hash    :
 go version  : go1.10.2
 go compiler : gc
 platform    : darwin/amd64
 features    : ImportDuringSolve=false
```
```s
$ go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/<my_name>/Library/Caches/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/<my_name>/go"
GORACE=""
GOROOT="/usr/local/Cellar/go/1.10.2/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.10.2/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/0x/0tyyz8f97gl2m8n0hp3n209m0000gn/T/go-build993181416=/tmp/go-build -gno-record-gcc-switches -fno-common"
```
```s
system_profiler SPSoftwareDataType

Software:

    System Software Overview:

      System Version: macOS 10.13.5 (17F77)
      Kernel Version: Darwin 17.6.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Computer Name: <my_computer_name>
      User Name: <my_user_name>
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 5:43
```

Gopkg.lock file (from dep)
```toml
# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.


[[projects]]
  name = "github.com/emirpasic/gods"
  packages = [
    "containers",
    "lists",
    "lists/arraylist",
    "trees",
    "trees/binaryheap",
    "utils"
  ]
  revision = "f6c17b524822278a87e3b3bd809fec33b51f5b46"
  version = "v1.9.0"

[[projects]]
  name = "github.com/gogo/protobuf"
  packages = [
    "gogoproto",
    "proto",
    "protoc-gen-gogo/descriptor",
    "sortkeys",
    "types"
  ]
  revision = "1adfc126b41513cc696b209667c8656ea7aac67c"
  version = "v1.0.0"

[[projects]]
  name = "github.com/golang/protobuf"
  packages = [
    "proto",
    "ptypes",
    "ptypes/any",
    "ptypes/duration",
    "ptypes/timestamp"
  ]
  revision = "b4deda0973fb4c70b50d226b1af49f3da59f5265"
  version = "v1.1.0"

[[projects]]
  branch = "master"
  name = "github.com/jbenet/go-context"
  packages = ["io"]
  revision = "d14ea06fba99483203c19d92cfcd13ebe73135f4"

[[projects]]
  name = "github.com/kevinburke/ssh_config"
  packages = ["."]
  revision = "9fc7bb800b555d63157c65a904c86a2cc7b4e795"
  version = "0.4"

[[projects]]
  branch = "master"
  name = "github.com/mitchellh/go-homedir"
  packages = ["."]
  revision = "3864e76763d94a6df2f9960b16a20a33da9f9a66"

[[projects]]
  name = "github.com/pachyderm/pachyderm"
  packages = [
    "src/client",
    "src/client/admin",
    "src/client/auth",
    "src/client/deploy",
    "src/client/enterprise",
    "src/client/health",
    "src/client/pfs",
    "src/client/pkg/config",
    "src/client/pkg/grpcutil",
    "src/client/pkg/pbutil",
    "src/client/pps",
    "src/client/version",
    "src/client/version/versionpb",
    "src/server/pkg/errutil"
  ]
  revision = "c3f9e2ebd030325f0dc576f812015c75172e9bf6"
  version = "v1.7.3"

[[projects]]
  name = "github.com/pelletier/go-buffruneio"
  packages = ["."]
  revision = "c37440a7cf42ac63b919c752ca73a85067e05992"
  version = "v0.2.0"

[[projects]]
  name = "github.com/satori/go.uuid"
  packages = ["."]
  revision = "f58768cc1a7a7e77a3bd49e98cdd21419399b6a3"
  version = "v1.2.0"

[[projects]]
  name = "github.com/sergi/go-diff"
  packages = ["diffmatchpatch"]
  revision = "1744e2970ca51c86172c8190fadad617561ed6e7"
  version = "v1.0.0"

[[projects]]
  name = "github.com/sirupsen/logrus"
  packages = ["."]
  revision = "c155da19408a8799da419ed3eeb0cb5db0ad5dbc"
  version = "v1.0.5"

[[projects]]
  name = "github.com/src-d/gcfg"
  packages = [
    ".",
    "scanner",
    "token",
    "types"
  ]
  revision = "f187355171c936ac84a82793659ebb4936bc1c23"
  version = "v1.3.0"

[[projects]]
  branch = "master"
  name = "github.com/xanzy/ssh-agent"
  packages = ["."]
  revision = "ba9c9e33906f58169366275e3450db66139a31a9"

[[projects]]
  branch = "master"
  name = "golang.org/x/crypto"
  packages = [
    "cast5",
    "curve25519",
    "ed25519",
    "ed25519/internal/edwards25519",
    "internal/chacha20",
    "openpgp",
    "openpgp/armor",
    "openpgp/elgamal",
    "openpgp/errors",
    "openpgp/packet",
    "openpgp/s2k",
    "poly1305",
    "ssh",
    "ssh/agent",
    "ssh/knownhosts",
    "ssh/terminal"
  ]
  revision = "df8d4716b3472e4a531c33cedbe537dae921a1a9"

[[projects]]
  branch = "master"
  name = "golang.org/x/net"
  packages = [
    "context",
    "http/httpguts",
    "http2",
    "http2/hpack",
    "idna",
    "internal/timeseries",
    "trace"
  ]
  revision = "1e491301e022f8f977054da4c2d852decd59571f"

[[projects]]
  branch = "master"
  name = "golang.org/x/sys"
  packages = [
    "unix",
    "windows"
  ]
  revision = "c11f84a56e43e20a78cee75a7c034031ecf57d1f"

[[projects]]
  name = "golang.org/x/text"
  packages = [
    "collate",
    "collate/build",
    "internal/colltab",
    "internal/gen",
    "internal/tag",
    "internal/triegen",
    "internal/ucd",
    "language",
    "secure/bidirule",
    "transform",
    "unicode/bidi",
    "unicode/cldr",
    "unicode/norm",
    "unicode/rangetable"
  ]
  revision = "f21a4dfb5e38f5895301dc265a8def02365cc3d0"
  version = "v0.3.0"

[[projects]]
  branch = "master"
  name = "google.golang.org/genproto"
  packages = ["googleapis/rpc/status"]
  revision = "81158efcc9f219c511e4d3c0d61a0e6e49c01a24"

[[projects]]
  name = "google.golang.org/grpc"
  packages = [
    ".",
    "balancer",
    "balancer/base",
    "balancer/roundrobin",
    "channelz",
    "codes",
    "connectivity",
    "credentials",
    "encoding",
    "encoding/proto",
    "grpclb/grpc_lb_v1/messages",
    "grpclog",
    "internal",
    "keepalive",
    "metadata",
    "naming",
    "peer",
    "resolver",
    "resolver/dns",
    "resolver/passthrough",
    "stats",
    "status",
    "tap",
    "transport"
  ]
  revision = "41344da2231b913fa3d983840a57a6b1b7b631a1"
  version = "v1.12.0"

[[projects]]
  name = "gopkg.in/src-d/go-billy.v4"
  packages = [
    ".",
    "helper/chroot",
    "helper/polyfill",
    "osfs",
    "util"
  ]
  revision = "df053870ae7070b0350624ba5a22161ba3796cc0"
  version = "v4.1.1"

[[projects]]
  name = "gopkg.in/src-d/go-git.v4"
  packages = [
    ".",
    "config",
    "internal/revision",
    "plumbing",
    "plumbing/cache",
    "plumbing/filemode",
    "plumbing/format/config",
    "plumbing/format/diff",
    "plumbing/format/gitignore",
    "plumbing/format/idxfile",
    "plumbing/format/index",
    "plumbing/format/objfile",
    "plumbing/format/packfile",
    "plumbing/format/pktline",
    "plumbing/object",
    "plumbing/protocol/packp",
    "plumbing/protocol/packp/capability",
    "plumbing/protocol/packp/sideband",
    "plumbing/revlist",
    "plumbing/storer",
    "plumbing/transport",
    "plumbing/transport/client",
    "plumbing/transport/file",
    "plumbing/transport/git",
    "plumbing/transport/http",
    "plumbing/transport/internal/common",
    "plumbing/transport/server",
    "plumbing/transport/ssh",
    "storage",
    "storage/filesystem",
    "storage/filesystem/internal/dotgit",
    "storage/memory",
    "utils/binary",
    "utils/diff",
    "utils/ioutil",
    "utils/merkletrie",
    "utils/merkletrie/filesystem",
    "utils/merkletrie/index",
    "utils/merkletrie/internal/frame",
    "utils/merkletrie/noder"
  ]
  revision = "57570e84f8c5739f0f4a59387493e590e709dde9"
  version = "v4.4.0"

[[projects]]
  name = "gopkg.in/warnings.v0"
  packages = ["."]
  revision = "ec4a0fea49c7b46c2aeb0b51aac55779c607e52b"
  version = "v0.1.2"

[solve-meta]
  analyzer-name = "dep"
  analyzer-version = 1
  inputs-digest = "a39403e546765aea0b2e17ce7f287a38a752f6a594275bf114c7570e09c491b0"
  solver-name = "gps-cdcl"
  solver-version = 1

```

**Comment by sjezewski** [comment]
So poking around here a bit.

It seems that this function call ... `uuid.NewV4()` returns 2 args at the version we have vendored (`36e9d2ebbde5e3f13ab2e25625fd453271d6522e`) but not at release `1.2.0` (`f58768cc1a7a7e77a3bd49e98cdd21419399b6a3`). Unfortunately the API at version `1.1.0` doesn't seem compatible either.

So there is a bit of impedance mismatch between these vendoring tools.

Since there is a way to specify an override using `dep` ... I've added this to my `Gopkg.toml` file:

```
 [[override]]
   name = "github.com/satori/go.uuid"
   revision="36e9d2ebbde5e3f13ab2e25625fd453271d6522e"
```

And that did the trick.
